core/server/commons/enclave_data_directory/enclave_data_directory.go

```diff
@@ -49,13 +50,19 @@ func (dir EnclaveDataDirectory) GetFilesArtifactStore() (*FilesArtifactStore, er
 		return nil, stacktrace.Propagate(err, "An error occurred ensuring the files artifact store dirpath '%v' exists.", absoluteDirpath)
 	}
 
+	var dbError error
 	// NOTE: We use a 'once' to initialize the filesArtifactStore because it contains a mutex,
 	// and we don't ever want multiple filesArtifactStore instances in existence
 	once.Do(func() {
-		currentFilesArtifactStore = newFilesArtifactStore(absoluteDirpath, relativeDirpath)
+		db, err := file_artifacts_db.GetOrCreateNewFileArtifactsDb()
+		if err != nil {
+			dbError = stacktrace.Propagate(err, "Failed to get file artifacts db")
+			return
+		}
+		currentFilesArtifactStore = newFilesArtifactStoreFromDb(absoluteDirpath, relativeDirpath, db)
 	})
 
-	return currentFilesArtifactStore, nil
+	return currentFilesArtifactStore, dbError
```

Comment:
I think that we should handle the error case before this return and return a nil object and the error, it's the current pattern when we return an error, objects should be nil